src/EventStore.Core.Tests/Services/Transport/Enumerators/Enumerator.AllSubscriptionFiltered.CombinationTests.cs

```diff
@@ -0,0 +1,533 @@
+				CheckpointType.AtZero => Position.Start,
+				CheckpointType.OneBeforeLast => GetPosition(_events[^2]),
+				CheckpointType.AtLast => GetPosition(_events[^1]),
+				CheckpointType.OneByteAfterLast => GetPositionPlusOneByte(_events[^1]),
+				CheckpointType.InvalidPosition => GetPositionMinusOneByte(_events[^1]),
+				_ => throw new ArgumentOutOfRangeException()
+			};
+
+			IEventFilter eventFilter = SubscriptionProps.EventFilterType switch {
+				EventFilterType.None => throw new ArgumentException(),
+				EventFilterType.StreamPrefix => EventFilter.StreamName.Prefixes(true, $"stream-{_testGuid}"),
+				EventFilterType.StreamRegex => EventFilter.StreamName.Regex(true, $"(.*?){_testGuid}(.*?)"),
+				EventFilterType.EventTypePrefix => EventFilter.EventType.Prefixes(true, $"type-{_testGuid}"),
+				EventFilterType.EventTypeRegex => EventFilter.EventType.Regex(true, $"(.*?){_testGuid}(.*?)"),
+				_ => throw new ArgumentOutOfRangeException()
+			};
+			return CreateAllSubscriptionFiltered(Node.Node.MainQueue, checkpoint, eventFilter, SystemAccounts.Anonymous);
+		}
+
+		private async Task<int> ReadExpectedEvents(EnumeratorWrapper sub, int nextEventIndex, int lastEventIndex, bool shouldFallBehindThenCatchUp = false) {
+			var fellBehind = false;
+			var caughtUp = false;
+
+			var numResponsesExpected = lastEventIndex - nextEventIndex + 1;
+			if (shouldFallBehindThenCatchUp)
+				numResponsesExpected += 2;
+
+			while (--numResponsesExpected >= 0) {
+				var response = await sub.GetNext();
+				switch (response) {
+					case Event evt:
+						var evtPos = _events[nextEventIndex++].OriginalPosition!.Value;
+						var evtTfPos = new TFPos(evtPos.CommitPosition, evtPos.PreparePosition);
+						Assert.AreEqual(evtTfPos, evt.EventPosition!.Value);
+						break;
+					case FellBehind:
+						if (!shouldFallBehindThenCatchUp)
+							Assert.Fail("Subscription fell behind.");
+
+						fellBehind = true;
+						break;
+					case CaughtUp:
+						if (!fellBehind)
+							Assert.Fail("Subscription caught up before falling behind");
+
+						if (!shouldFallBehindThenCatchUp)
+							Assert.Fail("Subscription fell behind then caught up.");
+
+						caughtUp = true;
+						break;
+					default:
+						Assert.Fail($"Unexpected response: {response}");
+						break ;
+				}
+			}
+
+			if (shouldFallBehindThenCatchUp) {
+				if (!fellBehind)
+					Assert.Fail("Subscription did not fall behind.");
+
+				if (!caughtUp)
+					Assert.Fail("Subscription fell behind but did not catch up.");
+			}
+
+			return nextEventIndex;
+		}
+
+		[SetUp]
+		public async Task SetUp() {
+			await WriteExistingEvents();
+			await PopulateExistingEvents();
+			_nextEventIndex = CalculateNextEventIndexFromCheckpoint();
+		}
+
+		[Test]
+		public async Task enumeration_is_correct() {
+			var sub = Subscribe();
+
+			if (SubscriptionProps.CheckpointType == CheckpointType.InvalidPosition) {
+				Assert.ThrowsAsync<ReadResponseException.InvalidPosition>(async () => await sub.GetNext());
```

Comment:
Do we really need an `async` lambda if we don't actually handle the inner value?

Reply:
@YoEight thanks, you're right, it doesn't look like it's needed. i suggest we leave it as is (for this PR) as it's in the tests and the same mistake has been repeated in the other enumerator tests